src/Components/WebAssembly/Build/test/BuildIntegrationTests/BuildIntegrationTest.cs

```diff
@@ -325,6 +325,35 @@ public async Task Build_WithI8NOption_CopiesI8NAssembliesWithLinkerEnabled()
             Assert.FileDoesNotExist(result, buildOutputDirectory, "wwwroot", "_framework", "_bin", "I18N.West.dll");
         }
 
+        [Fact]
+        public async Task Build_WithCustomOutputPath_Works()
+        {
+            // Arrange
+            using var project = ProjectDirectory.Create("standalone", additionalProjects: new[] { "razorclasslibrary" });
+
+            project.AddDirectoryBuildContent(
+@"<PropertyGroup>
+    <BaseOutputPath>$(MSBuildThisFileDirectory)build\bin\</BaseOutputPath>
+    <BaseIntermediateOutputPath>$(MSBuildThisFileDirectory)build\obj\</BaseIntermediateOutputPath>
+</PropertyGroup>");
+
+            var result = await MSBuildProcessManager.DotnetMSBuild(project, args: "/restore");
+            Assert.BuildPassed(result);
+
+#if DEBUG
+            const string configuration = "Debug";
+#else
+            const string configuration = "Release";
+#endif
+
+            var compressedFilesPath = Path.Combine(
+                project.DirectoryPath,
+                "build", "obj", configuration, "netstandard2.1",
```

Comment:
```suggestion
                "build",
                project.IntermediateOutputDirectory,
```

Reply:
The `IntermediateOutputDirectory` property is hardcoded to the default value. We can update this property from the test but, IMO, the trade-off of that vs listing out the path is small. Let me know if there is something I haven't considered here.

Reply:
Should have the right value no? https://github.com/dotnet/aspnetcore/blob/0dbff197214a029de8b47198e728cabe7a00783a/src/Components/WebAssembly/Build/test/BuildIntegrationTests/ProjectDirectory.cs#L157

Reply:
Synced offline. I missed that the `build/` prefix was still intact in your original suggestion. This makes sense to do.